includes/embeds/class-amp-scribd-embed-handler.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * Class AMP_SoundCloud_Embed_Handler
+ *
+ * @package AMP
+ * @since 1.3.1
+ */
+
+/**
+ * Class AMP_Scribd_Embed_Handler
+ */
+class AMP_Scribd_Embed_Handler extends AMP_Base_Embed_Handler {
+
+	/**
+	 * Registers embed.
+	 */
+	public function register_embed() {
+		add_filter( 'embed_oembed_html', [ $this, 'filter_embed_oembed_html' ], 10, 2 );
+	}
+
+	/**
+	 * Unregisters embed.
+	 */
+	public function unregister_embed() {
+		remove_filter( 'embed_oembed_html', [ $this, 'filter_embed_oembed_html' ], 10 );
+	}
+
+	/**
+	 * Filter oEmbed HTML for Scribd to to be AMP compatible.
+	 *
+	 * @param string $cache Cache for oEmbed.
+	 * @param string $url   Embed URL.
+	 * @return string Embed.
+	 */
+	public function filter_embed_oembed_html( $cache, $url ) {
+		if ( false === strpos( wp_parse_url( $url, PHP_URL_HOST ), 'scribd.com' ) ) {
+			return $cache;
+		}
+
+		$embed = $this->remove_script( $cache );
+		$embed = $this->inject_sandbox_attribute( $embed );
+
+		return $embed;
+	}
+
+	/**
+	 * Remove the 'script' element from the provided HTML if there is any.
+	 *
+	 * @param string $html HTML string.
+	 * @return string
+	 */
+	private function remove_script( $html ) {
+		$html_without_script = preg_replace( '#<script[^>].+?</script>#s', '', $html );
```

Comment:
Yes, the `[^>]` is not needed here indeed, and would only be used to start matching against attributes of the script tag.
However, with the rise of web components, we cannot rely too much on a given set of tag names anymore. So, the above would completely strip prepending elements whose tag name happens to start with `script`. So it would turn this:
```html
<script-highlighter>
   <script>alert();</script>
</script-highlighter>
```
into this, breaking the HTML:
```html
</script-highlighter>
```
Therefore, it would be safer to only match a strict `script` tag, with optional attributes that are separated by whitespace:
```suggestion
		$html_without_script = preg_replace( '#<script(?:\s.*?)?>.+?</script>#s', '', $html );
```